examples/tutorials/analysis-3d/non_detected_source.py

```diff
@@ -0,0 +1,251 @@
+# thus, the fitted model has a negative amplitude. We can use the
+# `~gammapy.modeling.select_nested_models` function to perform a likelihood ratio test to
+# see if this number is significant (See :doc:`docs/user-guide/howto.rst`).
+#
+
+LLR = select_nested_models(
+    Datasets(dataset), parameters=[skymodel.parameters["amplitude"]], null_values=[0]
+)
+print(LLR)
+
+######################################################################
+# You can see that the `ts ~ 4.7`, thus suggesting that the observed
+# fluctuations are not significant over the background. Note the here we have
+# only 1 free parameter, and thus, significance = sqrt(ts) ~ 2.2.
+# Now, we will estimate the differential upper limits of the source.
+
+###############################################################################
+# Differential upper limits
+# -------------------------
+#
+# Here, it is important to **set a reasonable model** on the dataset
+# before proceeding with the `~gammapy.estimators.FluxPointsEstimator` estimator. This model can come from
+# measurements from other instruments, be an extrapolation of the flux
+# observed at other wavelengths, come from theoretical estimations, etc. A
+# model with a negative amplitude as obtained above should not be
+# used.
+#
+# Note that the computed upper limits can depend on the spectral parameters of the assumed model.
+# Here, we compute the 3-simga upper limits for assuming a spectral index of 2.0.
+# We also fix the spatial parameters of the model to prevent the minimiser
+# from wandering off to different regions in the FoV.
+#
+
+
+model1 = skymodel.copy(name="model1")
+model1.parameters["amplitude"].value = 1e-14
+model1.parameters["index"].value = 2.0
+model1.freeze(model_type="spatial")
+
+energy_edges = dataset.geoms["geom"].axes["energy"].edges
+fp_est = FluxPointsEstimator(
+    selection_optional="all", energy_edges=energy_edges, n_sigma_ul=3
+)
+fp_est.norm.scan_min = 5
+fp_est.norm.scan_max = 100  # set this to a high value for large upper limits
+
+dataset.models = model1
+fp1 = fp_est.run(dataset)
+
+
+fp1.plot(sed_type="dnde")
+plt.show()
+
+
+######################################################################
+# Integral upper limits
+# ---------------------
+#
+# To compute the integral upper limits between certain energies,
+# we can simply run  `~gammapy.estimators.FluxPointsEstimator`
+# with one bin in energy
+
+emin = energy_edges[0]
+emax = energy_edges[-1]
+est2 = FluxPointsEstimator(selection_optional=["ul"], energy_edges=[emin, emax])
+fp2 = est2.run(dataset)
+print(
+    f"Integral upper limit between ${emin} and ${emax} is ${fp2.flux_ul.quantity.ravel()[0]}"
+)
+
+######################################################################
+# Note that this can be different the from correlated upper limits computed with the `ExcessMapEstimator`
+
+lima_maps.flux_ul.plot(add_cbar=True, cmap="viridis")
+plt.show()
+
```

Comment:
I think that one has a pedagogical issue here: the computed UL is about 10**-14, whereas the output of the excessmapestimator give 10**-12. This is very probably due to the different energy threshold (to be checked, by the way).
So, how we present the thing might confuse people.... One should maybe think to change of energy bin such that the results give the same order to magnitude.